python_homework/fastq_filtrator/fastq_filtrator.py

```diff
@@ -0,0 +1,219 @@
+# Simple fastq filtrator
+# run the .py file with flag -h in terminal for help
+
+import argparse
+
+
+def check_bound(bound):
+    """
+    Check type and number of elements in bounds.
+    """
+
+    if 2 < len(bound):
+        print('Error. Added more than two number to bounds.')
+        return
+
+    try:
+        normalized_bound = [float(value) for value in bound]
+        return normalized_bound
+
+    except ValueError:
+        print('Error. Not a number in bounds.')
+        return
+
+
+def make_len_bounds(length_bounds):
+    '''
+    Check types, number of values in length_bounds.
+    Transform it to [float, float].
+    '''
+
+    normalized_length_bound = check_bound(length_bounds)
+
+    if not normalized_length_bound:
+        return
+
+    # add lower bound to normalized_length_bounds
+    if len(normalized_length_bound) == 1:
+        return [0, normalized_length_bound[0]]
+
+    return normalized_length_bound
+
+
+def make_gc_bounds(gc_bounds):
+    '''
+    Check types, number of values in gc_bonus.
+    Transform it to [float, float].
+    '''
+
+    normalized_gc_bound = check_bound(gc_bounds)
+
+    if not normalized_gc_bound:
+        return
+
+    # add lower bound to normalized_gc_bounds
+    if len(normalized_gc_bound) == 1:
+        return [0, normalized_gc_bound[0]]
+
+    return normalized_gc_bound
+
+
+def filter_gc(seq_line, read_len, gc_bounds):
+    '''
+    Filter the read by GC content (in percent)
+    '''
+    gc_number = seq_line.count('C') + seq_line.count('G')
+    gc_content = 100 * gc_number / read_len
+
+    if gc_content < gc_bounds[0] or gc_content > gc_bounds[1]:
+        return False
+
+    return True
+
+
+def filter_length(read_len, length_bounds):
+    '''
+    Filter the read by length.
+    '''
+    if read_len < length_bounds[0] or read_len > length_bounds[1]:
+        return False
+
+    return True
+
+
+def filter_quality(read_len, quality_line, quality_threshold):
+    '''
+    Filter the read by quality
+    '''
+    quality = 0
+
+    for character in quality_line:
+        quality += (ord(character) - 33)
+
+    average_quality = quality / read_len
+
+    if average_quality < quality_threshold:
+        return False
+
+    return True
+
+
+def filter_fastq(lines, gc_bounds, length_bounds, quality_threshold):
+    '''
+    Check the quality for each read in fastq file.
+    If passed - return True, else - False.
+    '''
+    seq_line = lines[1][:-2]
+    quality_line = lines[3][:-2]
```

Comment:
Я так понял, что срез до 2 элемента с конца должен убирать перенос строки? Но тут есть несколько НО:
1. `\n` - это 1 символ, а не 2
2. Его может и не быть, например в последней строке файла
```suggestion
    seq_line = lines[1].strip()
    quality_line = lines[3].strip()
```
Так что целесеобразнее и универсальнее будет использование `strip`.
Из-за этого от последовательности отнимается один лишний символ и следовательно неправильно считаются все статистики.